### PR TITLE
feat: added support for scanning any account

### PR DESCRIPTION
**Pull request by elad-pticha**

### CODE REVIEW

**Comment by elad-pticha** [comment]
Changed `org` subcommand to `account`.
This feature allows us to determine if each account is a `user` or `organization` and to scan its GitHub Actions.

Now we iterate over each account:
``` python
for account in Config.account_name:
        generator = get_account_generator(account)

        for repo in generator:
            download_workflows_and_actions(repo)
```

and determine its type:
``` python
account_info = get_account_info(account_name=account_name)
    account_type = account_info.get("type")

    if account_type == "User":
        log.info(f"[+] Scanning user: {account_name}")
        return get_user_repository_generator(account_name)

    elif account_type == "Organization":
        log.info(f"[+] Scanning organization: {account_name}")
        return get_organization_repository_generator(account_name)

    else:
        log.error(f"[-] Failed to get account type for {account_name}")
        return None
```

From there, we continue the same.
This structure eliminates duplicate code and allows the user to scan in a single raven command for both organizations (like we had before) and user repositories.

Usage:
``` bash
raven download account --account-name ravendemo --account-name $PERSONAL_ACCOUNT --token $GITHUB_TOKEN
```

File: src/downloader/download.py [hunk 1]
```diff
@@ -24,8 +24,9 @@
 import src.logger.log as log
 
 
-def download_org_workflows_and_actions() -> None:
-    """Iterating all organization repositories through Github API.
+def download_account_workflows_and_actions() -> None:
+    """Firstly, we define it as an organization or a user account.
```
Comment (oreenlivnicode) [inline src/downloader/download.py:28]: *First

File: src/downloader/download.py [hunk 2]
```diff
@@ -37,11 +38,8 @@ def download_org_workflows_and_actions() -> None:
 
     We are trying to cache the downloads as much as we can to reduce redundant download attempts.
     """
-    log.debug("[+] Starting organization repository iterator")
-
-    for organization in Config.org_name:
-        log.debug(f"[+] Scanning {organization}")
-        generator = get_organization_repository_generator(organization)
+    for account in Config.account_name:
+        generator = get_account_generator(account)
```
Comment (oreenlivnicode) [inline src/downloader/download.py:42]: How about adding a debug massage: 
log.debug(f"[+] Scanning {account}")

**Review by oreenlivnicode — CHANGES_REQUESTED** [review]
I love that you can add multiple accounts to the same scan, but I am unsure about the obfuscation of the account type. It feels odd to me that the user doesn't have to explicitly specify whether the account is a user or an organization. 
What led to the decision not to ask for the account type explicitly?

File: src/downloader/download.py (hunk 2 above)
Comment (elad-pticha) [reply src/downloader/download.py:42]: There is a log inside the `get_account_generator` function:

``` python
if account_type == "User":
        log.info(f"[+] Scanning user: {account_name}")
        return get_user_repository_generator(account_name)

    elif account_type == "Organization":
        log.info(f"[+] Scanning organization: {account_name}")
        return get_organization_repository_generator(account_name)
```

**Comment by elad-pticha** [comment]
This solution allows users to scan different types of accounts without specifying if it is an organization or a user account. 
Requiring users to select the account type is unnecessary and will block us from scanning different versions in a single command.

I think this is the best solution as we don't care if we scan an organization or a user account as we want RAVEN to scan its repositories.

Command like this:
`raven download account --token $GITHUB_TOKEN --account-name userAccount --account-name microsoft`

This makes it as easy as possible to run RAVEN as you don't even have to check if a specific account is `org` type or `user` type, RAVEN just does this in the backend.

@oreenlivnicode WDYT?

**Comment by oreenlivnicode** [comment]
I got it @elad-pticha. I saw that github themselves use the terminology [account](https://docs.github.com/en/get-started/learning-about-github/types-of-github-accounts). So I am ok with it.

**Review by oreenlivnicode — APPROVED** [review]

**Pull request merged**